lib/contentful/client.rb

```diff
@@ -296,6 +298,16 @@ def sync(options = { initial: true })
 
     private
 
+    def ensure_select_is_doable(query)
+      return unless query.key?(:select)
+
+      query[:select] = query[:select].split(',').map(&:strip) if query[:select].is_a? String
+
+      query[:select] = query[:select].reject { |p| p.start_with?('sys.') }
```

Comment:
The library requires `sys.type` to function properly, while other SDKs require more parts of the `sys` block, as a team, we decided that instead of each SDK adding what they needed in order to function, we all enforced `sys` to provide consistency.

I remove the `sys.*` because they are irrelevant when the whole `sys` block is already returned.

I'll add a comment with the explanation.